src/scripts/mod.rs

```diff
@@ -0,0 +1,162 @@
+//! Scripting engine to run scripts based on tags and given ip-ports
+//! First read config file
+//!
+use anyhow::{anyhow, Result};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use which;
+
+// struct ScriptEngine {
+//     nmap: Bin,
+//     python: Bin,
+// }
+
+// impl ScriptEngine {
+//     fn build(&self, interpreter: String, arguments: Vec<String>, ip: String, ports: Vec<u16>) {
+
+//     }
+//     fn execute(script: Script, addr: &str, ports: Vec<_>, arguments: Vec<String>) {
+
+//     }
+// }
+
+pub struct Script {
```

Comment:
This is good, I can't imagine anything else we'd want to add.

Perhaps a specific argument for the language that will execude it?

```rust
language: String
```

So we can easily see if it's Python, Perl, Shell etc

Reply:
Parsed from the file itself, so it stay self-contained. We will have more possibilities to identify the interpreter, so it will be easy to fill that field.